### PR TITLE
docs: apex debugger intro update

### PR DESCRIPTION
**Pull request by sbudhirajadoc**
updated intro to add information about licensing requirements etc.

### What issues does this PR fix or reference?
#<Insert GitHub Issue>, @W-16041083@

### Functionality Before
No clear information about licensing requirement for each debugger type

### Functionality After
clearer information.

### CODE REVIEW

**Review by diyer — APPROVED** [review]
Looks great! Thanks Sonal

File: docs/_articles/en/apex/apex-debugging.md [hunk 1]
```diff
@@ -5,12 +5,17 @@ lang: en
...
+| [Replay Debugger](./en/apex/replay-debugger)       | None. Free.                                                                                                                                                                                                          | Easy to use. Use debug logs to “replay” events. Your debugger for most use cases.                                             |
+| [Interactive Debugger](./en/apex/interactive-debugger)  | - One free license in Performance and Unlimited Edition orgs.<br>- Available for purchase in Enterprise Edition orgs.<br>- Not available in Trial and Developer Edition orgs.<br>- Can be used in scratch orgs using the DebugApex feature. To use the debugger in a scratch org, the associated dev hub org must have the interactive debugger license. | A powerful debugger that lets you view and pause events in real time.                                                         |
```
Comment (jshackell-sfdc) [inline docs/_articles/en/apex/apex-debugging.md:12]:
```suggestion
| [Interactive Debugger](./en/apex/interactive-debugger)  | - One free license in Performance and Unlimited Edition orgs.<br>- Available for purchase in Enterprise Edition orgs.<br>- Not available in Trial and Developer Edition orgs.<br>- Can be used in scratch orgs using the DebugApex feature. To use the debugger in a scratch org, the associated Dev Hub org must have the interactive debugger license. | A powerful debugger that lets you view and pause events in real time.                                                         |
```

File: docs/_articles/en/apex/apex-debugging.md [hunk 2]
```diff
@@ -5,12 +5,17 @@ lang: en
...
+| [ISV Debugger](./en/apex/isv-debugger)          | - Available in subscriber sandbox orgs and scratch orgs.<br>- Can be used in scratch orgs using the DebugApex feature. To use the debugger in a scratch org, the associated dev hub org must have the interactive debugger license. For ISVs, this is their Partner Business Org (PBO).                                                                                                                                                                              | Use to debug managed package code directly in customer orgs in real time.              |
```
Comment (jshackell-sfdc) [inline docs/_articles/en/apex/apex-debugging.md:13]:
```suggestion
| [ISV Debugger](./en/apex/isv-debugger)          | - Available in subscriber sandbox orgs and scratch orgs.<br>- Can be used in scratch orgs using the DebugApex feature. To use the debugger in a scratch org, the associated Dev Hub org must have the interactive debugger license. For ISVs, this is their Partner Business Org (PBO).                                                                                                                                                                              | Use to debug managed package code directly in customer orgs in real time.              |
```

Comment (jshackell-sfdc) [reply docs/_articles/en/apex/apex-debugging.md:13]: Question: In the sentence "For ISVs, this is their Partner Business Org (PBO)", what is "this" referring to?  the associated Dev Hub org?  Or a Dev Hub org that has the license?  Is it possible to clarify?

File: docs/_articles/en/apex/apex-debugging.md (hunk 1 above)
Comment (jshackell-sfdc) [reply docs/_articles/en/apex/apex-debugging.md:12]: Also, [this page](https://developer.salesforce.com/tools/vscode/en/apex/interactive-debugger) calls the license "Apex Debugger License" (right?). I find Salesforce licenses SO confusing, so if both pages are talking about the same thing, I suggest you use the same terminology.  And maybe use the same link to https://help.salesforce.com/s/articleView?id=000382135&type=1.

File: docs/_articles/en/apex/apex-debugging.md (hunk 2 above)
Comment (jshackell-sfdc) [reply docs/_articles/en/apex/apex-debugging.md:13]: Same comment about interactive debugger license as above.

**Review by jshackell-sfdc — CHANGES_REQUESTED** [review]
See my suggestions.

**Review by jshackell-sfdc — APPROVED** [review]

File: docs/_articles/en/apex/apex-debugging.md (hunk 2 above)
Comment (sbudhirajadoc) [reply docs/_articles/en/apex/apex-debugging.md]: Its the same dev hub org. The associated dev hub org must have the license.

File: docs/_articles/en/apex/apex-debugging.md (hunk 1 above)
Comment (sbudhirajadoc) [reply docs/_articles/en/apex/apex-debugging.md]: Great catch! Let me confirm what the correct name for the license is and I'll update the doc.

**Pull request merged**